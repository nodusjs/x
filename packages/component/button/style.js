import { css } from "@nodusjs/std/dom";

export const style = (button) =>
  css`
    :host {
      width: ${button.onlyIcon ? `var(--button-width-${button.size})` : button.width};

      button {
        align-items: center;
        aspect-ratio: ${button.onlyIcon ? "1/1" : "auto"};
        background-color: var(--button-background-color-${button.color}-${button.variant});
        border: var(--button-border-${button.color}-${button.variant});
        border-radius: var(--radius-md);
        box-sizing: border-box;
        color: var(--button-color-${button.color}-${button.variant});
        cursor: pointer;
        display: inline-flex;
        font-family: var(--font-family-base);
        font-size: var(--button-font-size-${button.size});
        font-weight: var(--font-weight-semibold);
        gap: var(--button-gap-${button.size});
        height: var(--button-height-${button.size});
        justify-content: center;
        line-height: 1;
        padding: 0 ${button.onlyIcon ? 0 : "var(--spacing-lg)"};
        transition: all 0.2s ease-out;
        width: ${button.onlyIcon ? `var(--button-width-${button.size})` : button.width};

        &:hover {
          background-color: var(--button-background-color-${button.color}-${button.variant}_hover);
          color: var(--button-color-${button.color}-${button.variant}_hover);
        }

        &:focus-visible {
          box-shadow: var(--button-box-shadow-${button.color});
          outline: 0;
        }
      }
    }

    :host(:state(disabled)) {
      button {
        background-color: var(--bg-disabled);
        border: 1px solid var(--border-disabled_subtle);
        color: var(--fg-disabled);
        cursor: auto;
      }
    }

    :host(:state(hidden)) {
      display: none;
    }
  `;
