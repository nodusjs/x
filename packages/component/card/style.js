import { css } from "@nodusjs/std/dom";

export const style = (card) =>
  css`
    :host {
      background-color: var(--bg-primary);
      border: 1px solid var(--border-secondary);
      border-radius: var(--radius-xl);
      box-sizing: border-box;
      container-type: inline-size;
      display: inline-flex;
      height: ${card.height};
      padding: var(--spacing-${card.spacing});
      width: ${card.width};
      width: var(--width-${card.width}, ${card.width});
    }

    :host(:state(hidden)) {
      display: none;
    }
  `;
