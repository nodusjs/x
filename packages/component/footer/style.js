import { css } from "@nodusjs/std/dom";

/**
 * @function style
 *
 * @returns {CSSStyleSheet} A folha de estilo final para o rodapé.
 *
 * @description
 * Gera a folha de estilo (`CSSStyleSheet`) para o componente `x-footer`.
 *
 * Esta função define o layout principal do rodapé, estabelecendo-o como um
 * contêiner flexível de largura total com uma altura fixa e responsiva.
 * Ela também gerencia o estado de visibilidade do componente através da
 * pseudo-classe `:state(hidden)`.
 */
export const style = () =>
  css`
    :host {
      align-items: center;
      box-sizing: border-box;
      display: flex;
      height: 72px;
      width: 100%;

      @content (min-width: 768px) {
        height: 80px;
      }

      footer {
        container-type: inline-size;
        width: 100%;

        content {
          display: flex;
          gap: var(--spacing-2xl);
          justify-content: end;
        }
      }
    }

    :host(:state(hidden)) {
      display: none;
    }
  `;
