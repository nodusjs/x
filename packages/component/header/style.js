import { css } from "@nodusjs/std/dom";

export const style = () =>
  css`
    :host {
      align-items: center;
      box-sizing: border-box;
      display: flex;
      height: 72px;
      width: 100%;

      @content (min-width: 768px) {
        height: 80px;
      }

      header {
        container-type: inline-size;
        width: 100%;

        content {
          align-items: center;
          display: flex;
          gap: var(--spacing-2xl);
          justify-content: space-between;

          action {
            display: flex;
            gap: var(--spacing-2xl);
            justify-content: end;
          }
        }
      }
    }

    :host(:state(hidden)) {
      display: none;
    }
  `;
