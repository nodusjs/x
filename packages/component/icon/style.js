import { css } from "@nodusjs/std/dom";

export const style = (icon) =>
  css`
    :host {
      color: currentcolor;
      font-family: 'Material Symbols Outlined';
      font-style: normal;
      font-size: var(--icon-font-size-${icon.size});
      font-weight: normal;
      line-height: 1;
      letter-spacing: normal;
      text-transform: none;
      display: inline-block;
      white-space: nowrap;
      word-wrap: normal;
      direction: ltr;
      -moz-font-feature-settings: 'liga';
      -moz-osx-font-smoothing: grayscale;
    }
  `;
