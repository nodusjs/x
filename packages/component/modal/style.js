import { css } from "@nodusjs/std/dom";

/**
 * @function style
 *
 * @param {object} modal - A instância do componente `x-modal`.
 * @returns {CSSStyleSheet} A folha de estilo final para o modal.
 *
 * @description
 * Gera a folha de estilo (`CSSStyleSheet`) dinâmica para o componente `x-modal`.
 *
 * Esta função define os estilos para o contêiner principal (`:host`),
 * para a camada de sobreposição (`overflow`), e para o elemento `<dialog>`
 * que contém o conteúdo do modal. A visibilidade do modal é controlada
 * dinamicamente pela propriedade `modal.opened`.
 */
export const style = (modal) =>
  css`
    :host {
      box-sizing: border-box;
      display: ${modal.opened ? "block" : "none"};
      height: 100vh;
      position: fixed;
      width: 100vw;

      overflow {
        align-content: center;
        background-color: var(--bg-overlay) 0.7;
        backdrop-filter: var(--backdrop-blur-md);
        display: block;
        height: 100%;
        width: 100%;

        dialog {
          background-color: transparent;
          border: none;
          padding: 0;
        }
      }
    }
  `;
