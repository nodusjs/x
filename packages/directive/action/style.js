import { css } from "@nodusjs/std/dom";

/**
 * @function style
 *
 * @returns {CSSStyleSheet} A folha de estilo final para o contêiner de ação.
 *
 * @description
 * Gera a folha de estilo (`CSSStyleSheet`) para o componente `x-action`.
 *
 * Esta função define o layout do contêiner de ações, estabelecendo-o como
 * um contêiner flexível. Ela alinha os elementos de ação à direita
 * (`justify-content: end`) e aplica um espaçamento padrão entre eles
 * usando um token de espaçamento (`--spacing-2xl`).
 */
export const style = () =>
  css`
    :host {
      display: flex;
      gap: var(--spacing-2xl);
      justify-content: end;
    }
  `;
