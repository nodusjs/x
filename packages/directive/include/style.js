import { css } from "@nodusjs/std/dom";

export const style = () =>
  css`
    :host {
      box-sizing: border-box;
      container-type: inline-size;
      display: block;
    }
  `;
