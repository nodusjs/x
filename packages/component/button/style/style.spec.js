import { describe, it, expect } from 'vitest'
import { style } from './style.js'

describe('button style', () => {
  it('gera o CSS esperado com valores padrÃ£o', () => {
    const button = {
      color: 'brand',
      variant: 'solid',
      size: 'md',
      onlyIcon: false,
      width: 'auto',
    }

    const { cssRules: [{ cssText }]} = style(button)
    expect(cssText).toMatchSnapshot()
  })

  it('gera o CSS esperado com valores customizados e onlyIcon', () => {
    const button = {
      color: 'error',
      variant: 'ghost',
      size: 'xl',
      onlyIcon: true,
      width: '100px',
    }

    const { cssRules: [{ cssText }]} = style(button)
    expect(cssText).toMatchSnapshot()
  })
})
