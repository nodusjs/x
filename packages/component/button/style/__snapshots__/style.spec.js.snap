// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`button style > gera o CSS esperado com valores customizados e onlyIcon 1`] = `":host {  }"`;

exports[`button style > gera o CSS esperado com valores padrÃ£o 1`] = `":host {  }"`;
